src/Result.php

```diff
@@ -0,0 +1,95 @@
+<?php
```

Comment:
Is this a good Result object?

Reply:
I think so.  I would be fine checking it in.  I did suggest a couple changes below, but that could be implemented as subsequent PRs if you think they are helpful.

Reply:
:+1: